pkgs/development/python-modules/pyglet/default.nix

```diff
@@ -55,26 +62,39 @@ buildPythonPackage rec {
                 path = '${freetype}/lib/libfreetype${ext}'
             elif name[0:2] == 'av' or name[0:2] == 'sw':
                 path = '${ffmpeg-full}/lib/lib' + name + '${ext}'
+            elif name == 'openal':
+                path = '${openal}/lib/libopenal${ext}'
+            elif name == 'pulse':
+                path = '${libpulseaudio}/lib/libpulse${ext}'
+            elif name == 'Xi':
+                path = '${xorg.libXi}/lib/libXi${ext}'
+            elif name == 'Xinerama':
+                path = '${xorg.libXinerama}/lib/libXinerama${ext}'
+            elif name == 'Xxf86vm':
+                path = '${xorg.libXxf86vm}/lib/libXxf86vm${ext}'
             if path is not None:
                 return ctypes.cdll.LoadLibrary(path)
         raise Exception("Could not load library {}".format(names))
     EOF
   '';
 
-  propagatedBuildInputs = [ future ];
+  nativeBuildInputs = [ unzip ];
 
-  # needs an X server. Keep an eye on
-  # https://bitbucket.org/pyglet/pyglet/issues/219/egl-support-headless-rendering
+  # needs GL set up which isn't really possible in a build environment even in headless mode.
+  # tests do run and pass in nix-shell, however.
   doCheck = false;
 
   checkInputs = [
     pytest
   ];
 
+  # test list taken from .travis.yml
   checkPhase = ''
-    py.test tests/unit tests/integration
+    PYGLET_HEADLESS=True pytest --ignore=tests/base --ignore=tests/interactive --ignore=tests/integration --ignore=tests/unit/text/test_layout.py
```

Comment:
Please use pytestCheckHook.

Reply:
How do I pass the environment variable PYGLET_HEADLESS with pytestCheckHook?

Reply:
Figured this out, thanks for the suggestion.